api/endpoints/property.py

```diff
@@ -18,73 +35,129 @@ def get_properties():
         )
         db.session.add(prop_to_add)
         db.session.commit()
         return jsonify({"message": f"The property '{data['name']}' has been created."})
 
-    except ValidationError as err:
-        return jsonify({"ERROR": err.messages})
+    except ValidationError as err:  # Returns an error if a required field is missing
+        return jsonify({"error": err.messages})
 
 
-@property_endpoint.route('/api/property/update/<int:prop_id>', methods=['PUT'])
-def update_property(prop_id):
+@property_endpoint.route("/api/property/update", methods=["PUT"])
+@jwt_required
+def update_property():
     """
     Updates the details of the property in the database.
     """
-    data = request.get_json()
-    prop_to_update = Property.query.get(prop_id)
-    if not prop_to_update:
-        return jsonify({"error": f"The property with ID {prop_id} does not exist"})
+    data = request.form
+
+    user = get_jwt_identity()  # We get the username from token accessing this URL
+    owner = User.query.filter_by(
+        username=user
+    ).first()  # Then use the username from the token to query a user from db
+    prop_to_update = Property.query.get(data["id"])
+    if not owner:
+        return jsonify({"error": f"User not found"}), 403
+    elif not prop_to_update:
+        return jsonify({"message": "Invalid ID"})
+    elif prop_to_update.user_id != owner.id:
+        return jsonify({"message": "Action forbidden"}), 403
+
     try:
+        if "photos" not in data and request.files["photos"].filename == "":
+            return jsonify({"message": "You need to upload photos."})
+        # Check if the request data matches the schema else raise ValidationError
         prop_data = property_schema.load(data)
-        prop_to_update.name = prop_data['name']
-        prop_to_update.desc = prop_data['desc']
-        prop_to_update.price = prop_data['price']
-        prop_to_update.location = prop_data['location']
-        prop_to_update.photos = prop_data['photos']
+        image_files = request.files.getlist("photos")
+        images_folder = prop_to_update.image_folder
+        # Returns a list containing folder where images are saved and filenames
+        image_list = property_image_handler(image_files, images_folder)
+
+        image_list_to_json = json.dumps(image_list)
+
+        prop_to_update.name = prop_data["name"]
+        prop_to_update.desc = prop_data["desc"]
+        prop_to_update.price = prop_data["price"]
+        prop_to_update.location = prop_data["location"]
+        prop_to_update.photos = image_list_to_json
         prop_to_update.date = datetime.utcnow()
-        prop_to_update.user_id = prop_data['user_id']
+        prop_to_update.user_id = owner.id
         db.session.commit()
-        return jsonify({"message": "The details of the property has been updated."})
+        return (
+            jsonify({"message": "The details of the property has been updated."}),
+            201,
+        )
```

Comment:
same here lets show the json format of the user updated unlike the action message